### PR TITLE
AB#416642 -data-offline-store: Companies House data is pulled from offline store

### PR DESCRIPTION
**Pull request by RamakrishnaGujjula**
416642-data-offline-store: Companies House data is pulled from offline store

### CODE REVIEW

File: src/EPR.SubsidiaryBulkUpload.Application/Services/TableStorageProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ILogger<CompaniesHousCsvProcessor> logger) : ITableStorageProcessor
+{
+    private const string CurrentIngestion = "Current Ingestion";
+    private const string LatestCHData = "Latest CH Data";
+    private const string Latest = "Latest";
+    private readonly ILogger<CompaniesHousCsvProcessor> _logger = logger;
+
+    public async Task WriteToAzureTableStorage(IEnumerable<CompanyHouseTableEntity> records, string tableName, string partitionKey, string connectionString, int batchSize)
+    {
+        // Connect to the local Azure Table Storage
```
Comment (mikeymonster) [inline src/EPR.SubsidiaryBulkUpload.Application/Services/TableStorageProcessor.cs:18]: Comments should be removed to appease sonarqube

**Review by mikeymonster — DISMISSED** [review]
Approving without reviewing in sonarqube, because we are going to merge this into another active branch and review again.

File: src/EPR.SubsidiaryBulkUpload.Application/Services/TableStorageProcessor.cs (hunk 1 above)
Comment (RamakrishnaGujjula) [reply src/EPR.SubsidiaryBulkUpload.Application/Services/TableStorageProcessor.cs]: Done.

**Review by mikeymonster — APPROVED** [review]

File: src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHousCsvProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        csv.ReadHeader();
+        var header = csv.HeaderRecord;
+        _logger.LogInformation("Found csv header of length {Length} - {Values}", header.Length, string.Join(',', header));
+
+        while (await csv.ReadAsync())
+        {
+            rowCount++;
+
+            var field = csv.GetField(0);
+            if (rowCount % 1000 == 0)
```
Comment (BrianKnightEviden) [inline src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHousCsvProcessor.cs:34]: Why are we logging these this?
Wouldn't a simple log message of the number processed and the csv header do?

File: src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHousCsvProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Services;
+
+public class CompaniesHousCsvProcessor(
+    ILogger<CompaniesHousCsvProcessor> logger) : ICsvProcessor
+{
+    private readonly ILogger<CompaniesHousCsvProcessor> _logger = logger;
+
+    public async Task<int> ProcessStream(Stream stream)
```
Comment (BrianKnightEviden) [inline src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHousCsvProcessor.cs:14]: I don't think this method is being used. Can it be deleted?

Comment (mikeymonster) [reply src/EPR.SubsidiaryBulkUpload.Application/Services/CompaniesHousCsvProcessor.cs:14]: Leave this until we merge with the other branch code - the interface is used by both

**Review by BrianKnightEviden — APPROVED** [review]

**Pull request merged**